src/cpp/rtps/builtin/data/ProxyHashTables.hpp

```diff
@@ -48,6 +48,17 @@ class node_segregator
 
     using allocator_type = foonathan::memory::memory_pool<foonathan::memory::node_pool, RawAllocator>;
 
+#ifdef FOONATHAN_MEMORY_MEMORY_POOL_HAS_MIN_BLOCK_SIZE
+    node_segregator(
+            std::size_t nodes_to_allocate,
+            const bool& flag)
+        : block_size_(allocator_type::min_block_size(node_size, nodes_to_allocate ? nodes_to_allocate : 1))
```

Comment:
Children classes already take into account that nodes_to_allocate should be greater than 0
```suggestion
        : block_size_(allocator_type::min_block_size(node_size, nodes_to_allocate))
```